### PR TITLE
Add Version 8 support and fix ntohll

### PR DESCRIPTION
**Pull request by phrocker**
Begin version eight support. With this PR I can read RFiles produced from Java

### CODE REVIEW

File: src/examples/RfileReadExample.cpp [hunk 1]
```diff
@@ -201,8 +201,8 @@ main (int argc, char **argv)
         bigEndian = true;
     if (!IsEmpty (&outputFile))
     {
-        std::cout << "Writing test Rfile to " << outputFile << std::endl;
-        writeRfile (outputFile, bigEndian, 0);
+  //      std::cout << "Writing test Rfile to " << outputFile << std::endl;
```
Comment (phrocker) [inline src/examples/RfileReadExample.cpp]: need to fix

File: include/data/constructs/rfile/meta/IndexManager.h [hunk 2]
```diff
@@ -27,75 +27,61 @@
...
+  }
+  uint64_t
+  read(cclient::data::streams::InputStream *in);
 
-    std::shared_ptr<SerializedIndex>
-    lookup (std::shared_ptr<Key> key)
-    {
-    	std::shared_ptr< Block> block = std::make_shared<Block >(this, indexBlock);
-    	std::shared_ptr<SerializedIndex> index = std::make_shared<SerializedIndex> (block,block->lookup (key));
+  std::shared_ptr<SerializedIndex> lookup(std::shared_ptr<Key> key) {
```
Comment (phrocker) [inline include/data/constructs/rfile/meta/IndexManager.h]: const ref here

File: include/data/constructs/rfile/meta/IndexManager.h [hunk 3]
```diff
@@ -27,75 +27,61 @@
...
-        return size;
-    }
+  int getSize() {
+    return size;
+  }
 
-    std::shared_ptr<IndexBlock>
-    getIndexBlock (std::shared_ptr<IndexEntry > ie)
-    {
+  std::shared_ptr<IndexBlock> getIndexBlock(std::shared_ptr<IndexEntry> ie) {
```
Comment (phrocker) [inline include/data/constructs/rfile/meta/IndexManager.h]: const ref

File: include/data/constructs/rfile/bcfile/BlockCompressedFile.h [hunk 4]
```diff
@@ -171,8 +171,18 @@ class BlockCompressedFile : public cclient::data::streams::StreamInterface
...
         delete[] magicVerify;
         // get the index meta
-        in_stream->seek (fileLength - magic_size - VERSION_SIZE - 8);
-        offsetIndexMeta = in_stream->readLong ();
+        if (version.getMajor() == 1)
+        {
+          in_stream->seek (fileLength - magic_size - VERSION_SIZE - 8);
+          offsetIndexMeta = in_stream->readLong ();
+        }
+        else{
```
Comment (keith-turner) [inline include/data/constructs/rfile/bcfile/BlockCompressedFile.h]: Is this the main change to the persisted data?

Comment (phrocker) [reply include/data/constructs/rfile/bcfile/BlockCompressedFile.h]: There was one other place. Keep in mind that I already supported multi level indices and prefix encoding, so this was a very minor update. Will add some tests for different versions.

**Pull request merged**